src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -86,8 +86,8 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
     def ok_to_stop(self, daemon_ids: List[str], force: bool = False) -> HandleCommandResult:
         warn, warn_message = self._enough_daemons_to_stop(self.TYPE, daemon_ids, 'Grafana', 1)
         if warn and not force:
-            return HandleCommandResult(-errno.EBUSY, None, warn_message)
-        return HandleCommandResult(0, warn_message, None)
+            return HandleCommandResult(-errno.EBUSY, '', warn_message)
```

Comment:
weird.. so is this a proof that
```py
class HandleCommandResult(NamedTuple):
    retval: int = 0             # return code. E.g. 0 or -errno.EINVAL
    stdout: str = ""            # data of this result.
    stderr: str = ""            # Typically used for error messages.
```
is wrong?

Reply:
> weird.. so is this a proof that
> 
> ```python
> class HandleCommandResult(NamedTuple):
>     retval: int = 0             # return code. E.g. 0 or -errno.EINVAL
>     stdout: str = ""            # data of this result.
>     stderr: str = ""            # Typically used for error messages.
> ```
> 
> is wrong?

no, `HandleCommandResult` is right. `monitoring.py` was wrong

Reply:
but how come mypy is happy with monitoring.py?

Reply:
I just merged a PR that introduced the regression.

Reply:
#38854

Reply:
i see. the "make check" test for #38854 was performed 7 days ago before #38935 got merged.